xray/services/enrich.go

```diff
@@ -0,0 +1,99 @@
+package services
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-client-go/artifactory/services/utils"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+)
+
+const (
+	importGraph    = "api/v1/scan/import"
+	importGraphXML = "api/v1/scan/import_xml"
+)
+
+func (ss *ScanService) ImportGraph(scanParams XrayGraphImportParams) (string, error) {
+	httpClientsDetails := ss.XrayDetails.CreateHttpClientDetails()
+	var v interface{}
+	err := json.Unmarshal(scanParams.SBOMInput, &v)
+	utils.SetContentType("application/json", &httpClientsDetails.Headers)
+	url := ss.XrayDetails.GetUrl() + importGraph
+	if err != nil {
+		utils.SetContentType("application/xml", &httpClientsDetails.Headers)
+		url = ss.XrayDetails.GetUrl() + importGraphXML
+	}
+	requestBody := scanParams.SBOMInput
+	resp, body, err := ss.client.SendPost(url, requestBody, &httpClientsDetails)
+	if err != nil {
+		return "", err
+	}
+	if err = errorutils.CheckResponseStatusWithBody(resp, body, http.StatusOK, http.StatusCreated); err != nil {
+		scanErrorJson := ScanErrorJson{}
+		if e := json.Unmarshal(body, &scanErrorJson); e == nil {
+			return "", errorutils.CheckErrorf(scanErrorJson.Error)
+		}
+		return "", err
+	}
+	scanResponse := RequestScanResponse{}
+	if err = json.Unmarshal(body, &scanResponse); err != nil {
+		return "", errorutils.CheckError(err)
+	}
+	return scanResponse.ScanId, err
+}
+
+func (ss *ScanService) GetImportGraphResults(scanId string) (*ScanResponse, error) {
+	httpClientsDetails := ss.XrayDetails.CreateHttpClientDetails()
+	utils.SetContentType("application/json", &httpClientsDetails.Headers)
+
+	// The scan request may take some time to complete. We expect to receive a 202 response, until the completion.
+	endPoint := ss.XrayDetails.GetUrl() + scanGraphAPI
```

Comment:
is it suppost to be `scanGraphAPI`? or is it `importGraph` \ `importGraphXML` 
if scanGraphAPI, please add a comment about this.
If one of the other two, how can you make sure which endpoint you sent the request so you can access for results?

Reply:
scangraph api and added a comment